File: Personal/Java/BDD-POM/src/test/java/steps/Environment.java
package steps;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import io.cucumber.java.Before;
import io.cucumber.java.After;

public class Environment {

    private static WebDriver driver;

    @Before(order = 0)
    public void setup(){
        System.out.println("Global Hook");
        System.setProperty("webdriver.chrome.driver","/Users/gradulescu/Desktop/chromedriver");
        driver = new ChromeDriver();
    }


    @After(order=0)
    public void tearDown() throws Exception {
        System.out.println("After Hook");
        driver.quit();
        Thread.sleep(1000);
    }

    public static WebDriver getDriver(){
        return driver;
    }
}

File: Personal/Python/PythonAutomation/06 - BDD/features/environment.py
# environment.py e un fisier required de cucumber
import time
from selenium import webdriver

def before_all(context):
    context.browser=webdriver.Chrome("/Users/gradulescu/Desktop/Personal/Cursuri/ITF/Automation Course/Proiecte Python/RepositoryGithub/PythonAutomation/venv/04 - selenium_workshop_pom/resources/chromedriver")

def after_all(context):
    context.browser.quit()  # context  e un parametru caruia ii definesc anumite variabile ce sunt citite doar atunci cand avem nevoie
                            # cu acest context vom face legatura cu testele noastre
    time.sleep(3)